site/app/controllers/MiscController.php

```diff
@@ -83,6 +83,12 @@ private function checkValidAccess($is_zip) {
             // gradeable to get temporary info from
             // if team, get one of the user ids via the team id
             $current_gradeable = $this->core->getQueries()->getGradeable($path_gradeable_id, $current_user_id);
+            if($current_gradeable->getPeerGrading()) {
+                $peer_grade_set = $this->core->getQueries()->getPeerAssignment($path_gradeable_id, $current_user_id);
+                if(in_array($path_user_id, $peer_grade_set)) {
+                    return true;
+                }
+            }
```

Comment:
I feel like this if statement and its content should be moved down to be in the last if/else if/else part of checkValidAccess. It would also be best to return false so that it is similar in structure to the rest of this function. To make sure I'm understanding correctly, are peer graders allowed to download files of the student they are grading or is that not allowed?